roles/custom/matrix-synapse/defaults/main.yml

```diff
@@ -546,8 +546,16 @@ matrix_synapse_event_cache_size: "100K"
 
 # Controls cache sizes for Synapse.
 # Raise this to increase cache sizes or lower it to potentially lower memory use.
-# To learn more, see: https://github.com/matrix-org/synapse/issues/3939
-matrix_synapse_caches_global_factor: 0.5
+# To learn more, see:
+# - https://matrix-org.github.io/synapse/latest/usage/configuration/config_documentation.html#caching
+# - https://github.com/matrix-org/synapse/issues/3939
+matrix_synapse_caches_global_factor: 10
+matrix_synapse_caches_expire_caches: true
+matrix_synapse_caches_cache_entry_ttl: "30m"
+matrix_synapse_caches_sync_response_cache_duration: "2m"
+matrix_synapse_caches_autotuning_max_cache_memory_usage: "2048M"
```

Comment:
Oh yes this is a known problem with this PR and its addressed in the PR description maby not to well and has been mentioned in the matrix room when discussing this PR that this is indeed a good idea.

I will look at taking a stab at implementing this in the role as to fix this flaw. So Thanks for this comment.

Reply:
This should be resolved in the latest commits as we now use 1/8 and 1/16 as targets up to the previous values.

Reply:
After a conversation with @tcpipuk about the topic the conclusion has been reached that you dont rly want these numbers to go higher than X. Especially as these numbers are actually allowed to be reached per worker. And also the fact that its useless to allow this to scale to infinity as the cache factor creates its own max and autotune is essentially GC for that. 

And also that in the end all these values are sensible defaults that are intended to be good enough for those who dont touch these variables they are not intended to actually be a good idea to run for all users. Users who know what they are doing should be pushing way more advanced cache control settings than this PR has. Doing stuff like tuning the sizes of the specific types of caches etc and even preferably doing these settings per worker type or per worker.

Reply:
I personally would increase the "matrix_synapse_caches_cache_entry_ttl" because the 30m default is there for servers that aren't autotuning.

In an ideal world jemalloc would clear out anything that's not regularly used, so any objects that reach a long TTL would be accessed frequently, and it can really improve response times to let those live longer (especially on workers that do federation) but I guess it depends how far you want to deviate from the defaults?

I guess it could just be covered by a couple of lines in the docs to explain that setting 4-8h can massively improve response times on quieter servers, as the workers don't end up dumping everything when the user closes their client for an hour 🙂